lib/client/presence/presence.js

```diff
@@ -36,17 +37,21 @@ Presence.prototype.unsubscribe = function(callback) {
 };
 
 Presence.prototype.create = function(id) {
+  if (this._wantsDestroy) {
+    throw new Error('Presence is being destroyed');
+  }
   id = id || hat();
   var localPresence = this._createLocalPresence(id);
   this.localPresences[id] = localPresence;
   return localPresence;
 };
 
 Presence.prototype.destroy = function(callback) {
+  this._wantsDestroy = true;
   var presence = this;
+  var localIds = Object.keys(presence.localPresences);
```

Comment:
For future readers, it'd be worth a comment here to explain why this is gotten synchronously whereas the `remoteIds` is gotten inside the callback.